detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/HasPlatformType.kt

```diff
@@ -0,0 +1,77 @@
+package io.gitlab.arturbosch.detekt.rules.bugs
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import org.jetbrains.kotlin.descriptors.CallableDescriptor
+import org.jetbrains.kotlin.psi.KtCallableDeclaration
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.psiUtil.containingClassOrObject
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.types.isFlexible
+
+/**
+ * Platform types must be declared explicitly in public APIs to prevent unexpected errors.
+ *
+ * Based on code from Kotlin project:
+ * https://github.com/JetBrains/kotlin/blob/1.3.50/idea/src/org/jetbrains/kotlin/idea/intentions/SpecifyTypeExplicitlyIntention.kt#L86-L107
+ *
+ * <noncompliant>
+ * class Person {
+ *   fun apiCall() = System.getProperty("propertyName")
+ * }
+ * </noncompliant>
+ *
+ * <compliant>
+ * class Person {
+ *   fun apiCall(): String = System.getProperty("propertyName")
+ * }
+ * </compliant>
+ */
+
+class HasPlatformType(config: Config) : Rule(config) {
+
+    override val issue = Issue(
+        "HasPlatformType",
+        Severity.Maintainability,
+        "Platform types must be declared explicitly in public APIs.",
+        Debt.FIVE_MINS
+    )
+
+    override fun visitKtElement(element: KtElement) {
+        super.visitKtElement(element)
+
+        if (bindingContext == BindingContext.EMPTY) return
+        if (element is KtCallableDeclaration && element.hasImplicitPlatformType()) {
```

Comment:
The ifs can be merged easily.
```suggestion
        if (bindingContext != BindingContext.EMPTY && element is KtCallableDeclaration && element.hasImplicitPlatformType()) {
```